scripts/install_build_tools.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+set -e
+# shellcheck source=/dev/null
+. /etc/lsb-release
+echo "Distribution code name : ${DISTRIB_CODENAME}"
+apt-get update
+if [ "$DISTRIB_CODENAME" = "bionic" ];then 
+    echo "Installing bionic build tools ..."
+    DEPENDENCIES="build-essential \
+        cmake \
+        googletest \
+        google-mock \
+        gdb \
+        git"
+else
+    echo "Installing other linux distribution (focal or jammy) build tools ..."
+    DEPENDENCIES="build-essential \
+        cmake \
+        libgtest-dev \
+        libgmock-dev \
+        gdb \
+        git"
+fi
+apt-get install -y ${DEPENDENCIES}
```

Comment:
script lint suggests replacing this with`"${DEPENDENCIES[@]}"` but that results in `Bad substition`. This seems to apply to the exception for https://www.shellcheck.net/wiki/SC2086 but the work around does not work. I think its probably fine to keep like this since if/else statement means the variable has to be define. 
I think it doesn't work since I am using just a regular shell script and some of this might require a bash shell

Reply:
I think the issue is that you're still using a string for `DEPENDENCIES` when it should be a bash array. Try the following changes:

```
DEPENDENCIES=(
    build-essential 
    cmake 
    build-essential 
    libgtest-dev 
    libgmock-dev 
    gdb 
    git
)
```
```
DEBIAN_FRONTEND=noninteractive apt install --no-install-recommends --yes --quiet "${DEPENDENCIES[@]}"
```

Reply:
I think you will also need to change the shebang to `#!/bin/bash`.